pyserini/trectools/_base.py

```diff
@@ -0,0 +1,114 @@
+#
+# Pyserini: Python interface to the Anserini IR toolkit built on Lucene
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from copy import deepcopy
+from typing import Dict, List, Set
+from functools import cmp_to_key
+
+
+class TrecRunDoc:
```

Comment:
No sure why I se the need for both `TrecRun` and `TrecRunDoc`. Why not just add methods to `TrecRun` for reading from file and writing to file?

Reply:
On I get it, it the intent that `TrecRunDoc` is a single line in a `TrecRun`? Not sure I like this design...

Reply:
After thinking about this a bit, I'm more of the opinion that DF is the way to go... you've basically implemented what amounts to a "Row" in a DF here?

Reply:
Open to suggestions. `TrecRunDoc` is an abstraction to represent a document or each line in a TrecRun. We can print it as an object and it also takes care of the string serialization logic. 

It also works nicely to instantly create a placeholder or empty TrecRunDoc.